variables.tf

```diff
@@ -1,14 +1,34 @@
 # public subnet ip addresses CIDR blocks
 variable "public_subnet_cidr" {
+  description = "public subnet ip addresses CIDR blocks"
   default = [
     "10.0.0.0/24",
     "10.0.1.0/24"
   ]
 }
 # private subnet ip addresses CIDR blocks
 variable "private_subnet_cidr" {
+  description = "private subnet ip addresses CIDR blocks"
   default = [
     "10.0.2.0/24",
     "10.0.3.0/24"
   ]
 }
+variable "ami" {
+  description = "Ubuntu ami ID available in east-us-1"
+  default = "ami-09d56f8956ab235b3"
+}
```

Comment:
This will only work in us-east-1
To make it more universal, please use the aws_ami data source:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami

More examples here:
https://github.com/vladimirmukhin/data_source_example